src/Notifynder/Parsers/NotificationParser.php

```diff
@@ -20,14 +20,19 @@ class NotificationParser
     /**
      * Parse a notification and return the body text.
      *
-     * @param array|ModelNotification|BuilderNotification $notification
+     * @param ModelNotification $notification
      * @param int $categoryId
      * @return string
      * @throws ExtraParamsException
      */
     public function parse($notification, $categoryId)
     {
-        $category = NotificationCategory::findOrFail($categoryId);
+        $category = $notification->category;
+        if (is_null($category)) {
+            throw (new ModelNotFoundException)->setModel(
```

Comment:
Here you make the `$categoryId` param useless - I'm not 100% sure if this is ok. I think that we can remove it and see what happens - unittests & community.